exporter/exporterhelper/common.go

```diff
@@ -29,9 +32,37 @@ func NewDefaultTimeoutSettings() TimeoutSettings {
 
 // requestSender is an abstraction of a sender for a request independent of the type of the data (traces, metrics, logs).
 type requestSender interface {
+	start(ctx context.Context, host component.Host, set exporter.CreateSettings) error
+	shutdown()
 	send(req internal.Request) error
+	setNextSender(nextSender requestSender)
+}
+
+type baseRequestSender struct {
+	nextSender requestSender
+}
+
+var _ requestSender = (*baseRequestSender)(nil)
+
+func (b *baseRequestSender) start(context.Context, component.Host, exporter.CreateSettings) error {
+	return nil
+}
+
+func (b *baseRequestSender) shutdown() {}
+
+func (b *baseRequestSender) send(req internal.Request) error {
+	if b.nextSender != nil {
+		return b.nextSender.send(req)
+	}
+	return req.Export(req.Context())
```

Comment:
Very confusing this. If next is not initialized then exports it... Should we separate this into 2 structs on that always calls next and the other that calls export?

Reply:
If there is no next sender, it means it's the latest in the chain. This line is actually never called. The last sender `timeoutSender` explicitly calls `req.Export`. I left it here just to avoid potential panics in case the last sender is ever changed. I can remove it to avoid confusion.